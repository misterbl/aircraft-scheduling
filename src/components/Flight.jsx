import React, { PureComponent } from "react";
import rightArrow from "../assets/right-arrow.svg";

class Flight extends PureComponent {
  onSelectFlight = () => {
    const { flight, onSelectFlight, rotation, onRemoveFlight } = this.props;
    if (rotation) {
      onRemoveFlight(flight);
    } else {
      onSelectFlight(flight);
    }
  };

  render() {
    const { flight, rotation } = this.props;
    const {
      id,
      origin,
      readable_departure,
      destination,
      readable_arrival
    } = flight;

    return (
      <li className="flight" onClick={this.onSelectFlight}>
        <p className={`${rotation ? "" : "text-center"}`}>
          {`${rotation ? "Flight: " : ""}` + id}
        </p>
        <div className="d-flex justify-content-between">
          <div className="d-flex flex-column">
            <p>{origin}</p>
            <p>{readable_departure}</p>
          </div>
          {rotation && <img src={rightArrow} alt="right arrow" />}
          <div className="d-flex flex-column">
            <p>{destination}</p>
            <p>{readable_arrival}</p>
          </div>
        </div>
      </li>
    );
  }
}

export default Flight;
