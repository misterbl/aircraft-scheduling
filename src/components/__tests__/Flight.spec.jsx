import * as React from "react";
import { shallow } from "enzyme";
import Flight from "../Flight";

describe("Flight", () => {
  const props = {
    flight: {
      id: "1234",
      origin: "origin",
      readable_departure: "readable_departure",
      destination: "destination",
      readable_arrival: "readable_arrival"
    },
    onRemoveFlight: jest.fn(),
    onSelectFlight: jest.fn()
  };
  const wrapper = shallow(<Flight {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onSelectFlight if rotation is undefined", () => {
    const li = wrapper.find("li");
    li.simulate("click");
    expect(props.onSelectFlight).toHaveBeenCalled();
  });
  it("calls onRemoveFlight and show the righ arrow if rotation is true", () => {
    const propsWithRotation = { ...props, rotation: true };
    const wrapperWithRotation = shallow(<Flight {...propsWithRotation} />);
    const li = wrapperWithRotation.find("li");
    const img = wrapperWithRotation.find("img");
    li.simulate("click");
    expect(propsWithRotation.onRemoveFlight).toHaveBeenCalled();
    expect(img.length).toEqual(1);
  });
});
