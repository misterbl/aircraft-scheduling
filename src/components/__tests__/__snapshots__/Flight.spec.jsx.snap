// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Flight
    flight={
      Object {
        "destination": "destination",
        "id": "1234",
        "origin": "origin",
        "readable_arrival": "readable_arrival",
        "readable_departure": "readable_departure",
      }
    }
    onRemoveFlight={[MockFunction]}
    onSelectFlight={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="text-center"
        >
          1234
        </p>,
        <div
          className="d-flex justify-content-between"
        >
          <div
            className="d-flex flex-column"
          >
            <p>
              origin
            </p>
            <p>
              readable_departure
            </p>
          </div>
          <div
            className="d-flex flex-column"
          >
            <p>
              destination
            </p>
            <p>
              readable_arrival
            </p>
          </div>
        </div>,
      ],
      "className": "flight",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1234",
          "className": "text-center",
        },
        "ref": null,
        "rendered": "1234",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="d-flex flex-column"
            >
              <p>
                origin
              </p>
              <p>
                readable_departure
              </p>
            </div>,
            undefined,
            <div
              className="d-flex flex-column"
            >
              <p>
                destination
              </p>
              <p>
                readable_arrival
              </p>
            </div>,
          ],
          "className": "d-flex justify-content-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  origin
                </p>,
                <p>
                  readable_departure
                </p>,
              ],
              "className": "d-flex flex-column",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "origin",
                },
                "ref": null,
                "rendered": "origin",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "readable_departure",
                },
                "ref": null,
                "rendered": "readable_departure",
                "type": "p",
              },
            ],
            "type": "div",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  destination
                </p>,
                <p>
                  readable_arrival
                </p>,
              ],
              "className": "d-flex flex-column",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "destination",
                },
                "ref": null,
                "rendered": "destination",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "readable_arrival",
                },
                "ref": null,
                "rendered": "readable_arrival",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="text-center"
          >
            1234
          </p>,
          <div
            className="d-flex justify-content-between"
          >
            <div
              className="d-flex flex-column"
            >
              <p>
                origin
              </p>
              <p>
                readable_departure
              </p>
            </div>
            <div
              className="d-flex flex-column"
            >
              <p>
                destination
              </p>
              <p>
                readable_arrival
              </p>
            </div>
          </div>,
        ],
        "className": "flight",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1234",
            "className": "text-center",
          },
          "ref": null,
          "rendered": "1234",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="d-flex flex-column"
              >
                <p>
                  origin
                </p>
                <p>
                  readable_departure
                </p>
              </div>,
              undefined,
              <div
                className="d-flex flex-column"
              >
                <p>
                  destination
                </p>
                <p>
                  readable_arrival
                </p>
              </div>,
            ],
            "className": "d-flex justify-content-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    origin
                  </p>,
                  <p>
                    readable_departure
                  </p>,
                ],
                "className": "d-flex flex-column",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "origin",
                  },
                  "ref": null,
                  "rendered": "origin",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "readable_departure",
                  },
                  "ref": null,
                  "rendered": "readable_departure",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    destination
                  </p>,
                  <p>
                    readable_arrival
                  </p>,
                ],
                "className": "d-flex flex-column",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "destination",
                  },
                  "ref": null,
                  "rendered": "destination",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "readable_arrival",
                  },
                  "ref": null,
                  "rendered": "readable_arrival",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
